src/coordinates/uniform_cartesian.hpp

```diff
@@ -269,6 +269,52 @@ class UniformCartesian {
     return 0.0;
   }
 
+  template <class... Args>
+  KOKKOS_FORCEINLINE_FUNCTION Real Volume(CellLevel cl, TopologicalElement el,
+                                          Args... args) const {
+    using TE = TopologicalElement;
+    if (cl == CellLevel::same) {
+      if (el == TE::CC) {
+        return cell_volume_;
+      } else if (el == TE::F1) {
+        return area_[X1DIR - 1];
+      } else if (el == TE::F2) {
+        return area_[X2DIR - 1];
+      } else if (el == TE::F3) {
+        return area_[X3DIR - 1];
+      } else if (el == TE::E1) {
+        return dx_[X1DIR - 1];
+      } else if (el == TE::E2) {
+        return dx_[X2DIR - 1];
+      } else if (el == TE::E3) {
+        return dx_[X3DIR - 1];
+      } else if (el == TE::NN) {
+        return 1.0;
+      }
```

Comment:
Is this all new or can we call the older volume method which defaults to `CellLevel::same`?

Reply:
the old `Volume` method is templated on the topological element. Here I added a new `Volume` overload that is not templated and takes a topological type and a cell level as an argument. It should give the same results as the templated volume function.

Reply:
What I meant was that for the `CellLevel::same` branch can you call the old method?

Reply:
You could, but since the old one is templated on TE and this one is not it doesn't save any lines of code since you would have to call each template by hand. I am not actually sure that this really needs to be templates though.

Reply:
Oh I see. Could the old one optionally take a runtime? That might be useful...